defaults/initrd.scripts

```diff
@@ -2068,8 +2146,19 @@ openLUKS() {
 			fi
 
 			# At this point, {header,key}file or not, we're ready!
-			crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}"
-			crypt_filter_ret=$?
+            if [ "${PLYMOUTH}" = '1' ]
+            then
+                ply_cmd="${ply_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}"
+                local ply_cmd_file="$(mktemp -t "ply_cmd.XXXXXX")"
+                printf '#!/bin/sh\n%s\n' "${ply_cmd}" > "${ply_cmd_file}"
+                run chmod 500 "${ply_cmd_file}"
+            fi
+            crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}" "${ply_cmd_file}"
```

Comment:
Please create separate `crypt_filter_ply()` (or smth. similar) and place it inside condition block above. Or declare `local ply_cmd_file=""` near of `local ply_cmd=""`. Also use tabs instead of spaces.

Reply:
What happens if the `LUKS_KEY` is defined? Will `plymouth ask-for-password` show a password entry dialog and skip right away?

Reply:
Will push the changes once I get home.

I'm not quite sure what you're asking about the LUKS_KEY.  If it is a gpg key, then plymouth shows a password entry dialog and you have to enter your key decryption password correctly before it continues.  I have not tested the case where a non-gpg, non-password-protected key is used, but I presume you could just hit enter on the empty password field to continue.

Reply:
I think it is necessary to prevent handling by plymouth if non-gpg key is used. Show a password dialog only for missing key and gpg-key.

Reply:
Please ignore my first comment ab. `crypt_filter_ply()`. I suggest the following `crypt_filter()`:
```
crypt_filter() {
	[ -z "${LUKS_KEY}" ] || [ "$(echo ${LUKS_KEY} | grep -o '.gpg$')" = ".gpg" ]
	local ply_ask=$?

	if [ "${CRYPT_SILENT}" = '1' ]
	then
		eval run ${1}
	elif [ "${PLYMOUTH}" = '1' ] && plymouth_running && [ ply_ask -eq 0 ]
	then
		local ply_cmd_file="$(mktemp -t "ply_cmd.XXXXXX")"
		printf '#!/bin/sh\n%s\n' "${1}" > "${ply_cmd_file}"
		run chmod 500 "${ply_cmd_file}"
		run plymouth ask-for-password --prompt "Enter LUKS passphrase" \
			--number-of-tries=3 --command="${ply_cmd_file}"
		res=$?
		run rm "${ply_cmd_file}"
		return ${res}
	else
		splashCmd 'verbose' >/dev/null &
		eval run ${1}
		res=$?
		if [ ${res} -eq 0 ]
		then
			splashCmd set_msg 'Disk unlocked.'
		fi
		return ${res}
	fi
}
```

Reply:
I like this solution; it keeps everything strictly local.  Will push it after I test on my system.